adapter/cmsis/BUILD.gn

```
@@ -81,3 +81,127 @@ run_qemu_check("run_cmsis_unittest") {
     checker = "//kernel/adapter/unittests.checker"
   }
 }
+
+validation_deps = adapter_deps
+
+_sysroot = exec_script("//build/scripts/get_sysroot.py",
+                       [ "arm-none-eabi-gcc" ],
+                       "string")
+
+build_rust("adapter_validation") {
+  testonly = true
+  crate_name = "adapter_validation"
+  crate_type = "bin"
+  sources = [ "src/validation.rs" ]
+  edition = "2021"
+  proc_macro_deps = [
+    "//external/vendor/delegate-0.13.4:delegate",
+    "//kernel/test_harness:blueos_test_macro",
+  ]
+  deps = validation_deps
+  deps += [
+    ":cmsis_adapter",
+    ":validation_cases",
+    "//external/vendor/semihosting-0.1.20:semihosting",
+
+    # in fact, validation_cases need an c library
+    # "//librs:librs",
+  ]
+  if (board == "qemu_mps2_an385") {
+    deps += [ "//external/vendor/cortex-m-0.7.7:cortex_m" ]
+  }
+  configs += [
+    "//kernel/kconfig:kconfigs",
+    "//build/boards/${board}:kernel_config",
+  ]
+  rustflags = [ "--test" ]
+}
+
+run_clippy("adapter_validation_clippy") {
+  testonly = true
+  deps = [ ":adapter_validation" ]
+}
+
+static_library("validation_cases") {
+  testonly = true
+  sources = [
```

```suggestion
  sources = _enabled_cases + _disabled_cases
```
Please classify these cases into lists.

---

this is not neccesary, enable cases only controlled by 'config.h'